integration/alertmanager_test.go

```diff
@@ -40,3 +49,56 @@ func TestAlertmanager(t *testing.T) {
 	// Ensure no service-specific metrics prefix is used by the wrong service.
 	assertServiceMetricsPrefixes(t, AlertManager, alertmanager)
 }
+
+func TestAlertmanagerStoreAPI(t *testing.T) {
+	s, err := e2e.NewScenario(networkName)
+	require.NoError(t, err)
+	defer s.Close()
+
+	minio := e2edb.NewMinio(9000, AlertmanagerS3Flags["-alertmanager.storage.s3.buckets"])
+	require.NoError(t, s.StartAndWaitReady(minio))
+
+	am := e2ecortex.NewAlertmanager(
+		"alertmanager",
+		mergeFlags(
+			AlertmanagerFlags,
+			AlertmanagerS3Flags,
+		),
+		"",
+	)
+
+	require.NoError(t, s.StartAndWaitReady(am))
+	require.NoError(t, am.WaitSumMetrics(e2e.Equals(0), "cortex_alertmanager_configs"))
+
+	c, err := e2ecortex.NewClient("", "", am.HTTPEndpoint(), "", "user-1")
+	require.NoError(t, err)
+
+	_, err = c.GetAlertmanagerConfig(context.Background())
+	require.Error(t, err)
+	require.EqualError(t, err, e2ecortex.ErrNotFound.Error())
+
+	err = c.SetAlertmanagerConfig(context.Background(), cortexAlertmanagerUserConfigYaml, map[string]string{})
+	require.NoError(t, err)
+
+	time.Sleep(time.Second * 2)
```

Comment:
I would prefer to rely on `time.Sleep()`. Would we get the same if we do `require.NoError(t, am.WaitSumMetrics(e2e.Equals(1), "cortex_alertmanager_configs"))`?